app/models/request.rb

```diff
@@ -16,4 +16,8 @@ def check_status
     rejected! if receivers.any?(&:rejected?)
   end
 
+  def delete_receivers
+    receivers.each(&:deleted!)
```

Comment:
if you do `dependent: :destroy`, you won't have to do this.
check association in active record.